gprofiler/main.py

```diff
@@ -463,6 +503,13 @@ def parse_cmd_args() -> configargparse.Namespace:
         default=False,
         help="Whether to upload the profiling results to the server",
     )
+    parser.add_argument(
```

Comment:
Did you try adding it as a subcommand? I think it makes more sense UX wise, and also allows us to structure the code better. With the current way this is written, the preconditions for gProfiler run are still checked in the upload mode, despite there's no need for that (checking for root, checking for init PID NS, ...).

What I had in mind is - `./gprofiler upload-file` will be a subcommand accepting only the relevant parameters - let's begin with `token`, `service-name` and later if needed we can add others such as `server-upload-timeout`. It will reject other parameters because they are truly not used. `./gprofiler` will run the default behavior (i.e what we currently have) and accept all arguments.
See [this](https://stackoverflow.com/questions/6365601/default-sub-command-or-handling-no-sub-command-with-argparse) SO question with an example of how to do that with `argparse`, and then there's a way to define the argument `--token` once, and have it available both for the default parser (`./gprofiler`) and for the upload subparser (`./gprofiler upload-file`)

Reply:
Good now :)